### DISCUSSION TITLE
Character is getting sticky at edges

### DISCUSSION THREAD

**Discussion by LeXXik** [jrouwe/JoltPhysics.js#82] [Q&A] [answered]
I'm trying to figure out why my character is getting sticky at the edges. The character should be similar to the repo example. In the clip I release keyboard once it gets stuck, so it should free fall. In the repo example it doesn't exhibit this, so it must be my spaghetti code. I've disabled any sort of callbacks to exclude their case. Any idea where I could debugging this?

https://github.com/jrouwe/JoltPhysics.js/assets/5677782/b8dc4146-b73d-44ea-980c-2b92844874aa

**Comment by jrouwe** [comment]
Looks like you're not setting CharacterVirtualSetting.mSupportingVolume correctly, see comment:

```
	/// Plane, defined in local space relative to the character. Every contact behind this plane can support the
	/// character, every contact in front of this plane is treated as only colliding with the player.
	/// Default: Accept any contact.
```

**Reply by LeXXik** [reply, marked as answer]
Hmm, am I correct in this line of thought: if the capsule height is 2 meters, then I should create a plane that is pointing up at distance -1 meters, which places the plane at the character's feet.

I think I do it here:
![image](https://github.com/jrouwe/JoltPhysics.js/assets/5677782/340d9483-0320-4195-8786-d446a0b6097f)

**Reply by LeXXik** [reply]
Ah, you are correct. I forgot that the shape is decorated. It is offset by half height, so the plane distance is no longer correct. After taking offset into account, it works correctly. Thank you!

**Reply by LeXXik** [reply]
Actually, the repo example doesn't do it (and adding that offset breaks walking on slopes). Hmm. Must be something else then.

**Reply by LeXXik** [reply]
Yeah, in the end it was my shape's original position. It was a bit low. Works correctly now. Thanks again!

**Reply by jrouwe** [reply]
> Actually, the repo example doesn't do it (and adding that offset breaks walking on slopes). Hmm. Must be something else then.

Not sure what you mean by 'repo example', but the character virtual demo does set it:

https://github.com/jrouwe/JoltPhysics.js/blob/1c24480a4c1dd3f0dfc6e58171b51dfbac0800ed/Examples/character_virtual.html#L314

**Reply by LeXXik** [reply]
Yep, correct, I just worded it badly. English is hard :D Nevermind.
